pkg/generate.go

```diff
@@ -56,15 +69,45 @@ func Generate(fs afero.Afero, gitOpsFolder string, outputFolder string, componen
 		deploymentFileName: deployment,
 	}
 
+	var service *corev1.Service
+	var route *routev1.Route
 	// If a targetPort was specified, also generate a service and route
-	if component.TargetPort != 0 {
-		service := generateService(component)
-		route := generateRoute(component)
+	if len(component.KubernetesResources.Services) == 0 && len(component.KubernetesResources.Routes) == 0 && component.TargetPort != 0 {
```

Comment:
So clients of this API will always provide both a service and route?  In one case in the 'normal or regular' generation scenario (Prior to this PR), a route is not generated if the TargetPort is not 0 (component.TargetPort), so then you will have a situation where there is a service but not a route.

Reply:
but we are generating in the main branch - check here pls https://github.com/redhat-developer/gitops-generator/blob/main/pkg/generate.go#L60-L62

Reply:
Yeah, this is fine.